unit/analyses/call_graph.cpp

```diff
@@ -0,0 +1,109 @@
+
+#include <iostream>
+
+#include <catch.hpp>
+
+#include <analyses/call_graph.h>
+
+#include <util/symbol_table.h>
+#include <util/std_code.h>
+
+#include <goto-programs/goto_convert_functions.h>
+
+static symbolt create_void_function_symbol(
+  const irep_idt &name,
+  const codet &code)
+{
+  code_typet void_function_type;
+  symbolt function;
+  function.name=name;
+  function.type=void_function_type;
+  function.mode=ID_java;
+  function.value=code;
+  return function;
+}
+
+static bool multimap_key_matches(
+  const std::multimap<irep_idt, irep_idt> &map,
+  const irep_idt &key,
+  const std::set<irep_idt> &values)
+{
+  auto matching_values=map.equal_range(key);
```

Comment:
This is fine - but you can actually use the `REQUIRE` macro in here too which allows the errors to be a bit more informative.